ios/FluentUI/Vnext/PersonaButtonCarousel/PersonaButtonCarousel.swift

```diff
@@ -5,9 +5,79 @@
 
 import SwiftUI
 
+/// `MSFPersonaCarouselState` contains PersonaCarousel properties
+///
+/// - `buttonSize`: specifies whether to use small or large avatars
+/// - `onTapAction`: provides tap gesture
+@objc public protocol MSFPersonaCarouselState {
+    var buttonSize: MSFPersonaButtonSize { get }
+    var onTapAction: ((_ personaButtonState: MSFPersonaButtonState, _ index: Int) -> Void)? { get set }
+}
+
+/// Properties that make up PersonaGrid content
+class MSFPersonaCarouselStateImpl: NSObject, ObservableObject, Identifiable, MSFPersonaCarouselState {
+    let tokens: MSFPersonaButtonCarouselTokens
+    let buttonSize: MSFPersonaButtonSize
+
+    @Published var onTapAction: ((_ personaButtonState: MSFPersonaButtonState, _ index: Int) -> Void)?
+
+    @Published var buttons: [MSFPersonaButtonStateImpl] = [] {
+        didSet {
+            // When the buttons array changes, any new entries need to have their action set
+            // to pass through to the main callback for the carousel
+            let old = Set(oldValue)
+            let new = Set(buttons)
+            let newElements = new.subtracting(old)
+
+            newElements.forEach { button in
```

Comment:
How about adding an initializer to the PersonaButton that takes an action closure and pass the PersonaCarouselState closure that will call the client code handler for the Carousel tap event?

Reply:
Now we're thinking with SwiftUI! Awesome idea 👍